### PR TITLE
[fix](array)(parquet) fix be core dump due to load from parquet file containing array types

### LINKED ISSUE #17294: [Bug] load from parquet file containing array types leads to be core dump
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

```
*** Current BE git commitID: 62ec74f4e ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 70908 (TID 71775 OR 0x7f7b111d3700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/common/signal_handler.h:428
 1# 0x00007F7C6E913500 in /lib64/libpthread.so.0
 2# std::_Bit_reference::operator bool() const at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/stl_bvector.h:87
 3# std::_Bit_const_iterator::operator*() const at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/stl_bvector.h:348
 4# std::_Bit_const_iterator::operator[](long) const at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/stl_bvector.h:396
 5# std::vector<bool, std::allocator<bool> >::operator[](unsigned long) const at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/stl_bvector.h:911
 6# doris::vectorized::DataTypeFactory::create_data_type(doris::TypeDescriptor const&, bool) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/vec/data_types/data_type_factory.cpp:168
 7# doris::vectorized::VFileScanner::_init_src_block(doris::vectorized::Block*) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/vec/exec/scan/vfile_scanner.cpp:243
 8# doris::vectorized::VFileScanner::_get_block_impl(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/vec/exec/scan/vfile_scanner.cpp:173
 9# doris::vectorized::VScanner::get_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/vec/exec/scan/vscanner.cpp:61
10# doris::vectorized::ScannerScheduler::_scanner_scan(doris::vectorized::ScannerScheduler*, doris::vectorized::ScannerContext*, doris::vectorized::VScanner*) at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/vec/exec/scan/scanner_scheduler.cpp:306
11# doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}::operator()() const at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/vec/exec/scan/scanner_scheduler.cpp:192
12# void std::__invoke_impl<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}&>(std::__invoke_other, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}&) at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/invoke.h:61
13# std::enable_if<is_invocable_r_v<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}&>, void>::type std::__invoke_r<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}&>(doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}&) at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/invoke.h:117
14# std::_Function_handler<void (), doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const::{lambda()#3}>::_M_invoke(std::_Any_data const&) at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/std_function.h:292
15# std::function<void ()>::operator()() const at /home/disk3/lzl/icode/baidu/bdg/doris/palo-toolchain/ldb_toolchain/include/c++/11/bits/std_function.h:560
16# doris::FunctionRunnable::run() at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/util/threadpool.cpp:46
17# doris::ThreadPool::dispatch_thread() at /home/disk3/lzl/icode/baidu/bdg/doris/core/be/src/util/threadpool.cpp:537
```

### What You Expected?

load successfully

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #17294

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
run buildall

File: be/src/vec/exec/format/parquet/schema_desc.cpp [hunk 1]
```diff
@@ -136,6 +136,8 @@ Status FieldDescriptor::parse_node_field(const std::vector<tparquet::SchemaEleme
         transform(t_schema.name.begin(), t_schema.name.end(), lower_case_name.begin(), ::tolower);
         node_field->name = lower_case_name;
         node_field->type.type = TYPE_ARRAY;
+        node_field->type.children.push_back(child->type);
```
Comment (cambyzju) [inline be/src/vec/exec/format/parquet/schema_desc.cpp:139]: we already add functions in pr: https://github.com/apache/doris/pull/16978, please use it.

Comment (luozenglin) [reply be/src/vec/exec/format/parquet/schema_desc.cpp]: done, thanks

**Comment by luozenglin** [comment]
run beut

**Comment by morningman** [comment]
@AshinGau PTAL

**Comment by luozenglin** [comment]
run buildall

File: be/src/runtime/types.cpp [hunk 2]
```diff
@@ -264,9 +264,9 @@ TypeDescriptor::TypeDescriptor(const google::protobuf::RepeatedPtrField<PTypeNod
     }
 }
 
-void TypeDescriptor::add_sub_type(TypeDescriptor&& sub_type, bool&& is_nullable) {
-    children.emplace_back(sub_type);
-    contains_nulls.emplace_back(is_nullable);
+void TypeDescriptor::add_sub_type(TypeDescriptor sub_type, bool is_nullable) {
```
Comment (AshinGau) [inline be/src/runtime/types.cpp:267]: `OrcReader` has used this method, and use right values as parameters.

Comment (luozenglin) [reply be/src/runtime/types.cpp:267]: This function can match right values, and if the argument passed in is a right value, the overhead is two move operations

**Comment by luozenglin** [comment]
run buildall

File: be/src/runtime/types.cpp (hunk 2 above)
Comment (AshinGau) [reply be/src/runtime/types.cpp:267]: OK, it looks well. How about another polymorphic function `add_sub_type(TypeDescriptor&& sub_type, std::string&& field_name, bool&& is_nullable)`? It feels strange that the parameter forms of the two functions are not uniform.

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
This bug only affects the master branch and I have removed the 1.2 label@morningman

**Comment by AshinGau** [comment]
LGTM

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**